work_plans/init_seed_data.md

```diff
@@ -0,0 +1,489 @@
+├── calculate_previous_month() → returns (start_date, end_date)
+├── fetch_players_data() → writes players.csv
+├── fetch_teams_data() → writes teams.csv
+├── fetch_games_data() → writes games.csv
+│   ├── Gets all teams
+│   ├── For each team, finds games in date range
+│   └── Deduplicates and aggregates all games
+└── fetch_boxscores_data() → writes boxscores.csv
+    ├── Uses game IDs from games.csv
+    ├── For each game, fetches player boxscores
+    └── Aggregates all player stats
+
+main()
+├── Parse CLI arguments
+├── Calculate date range
+├── Fetch all data types
+└── Report summary
+```
+
+### Date Range Calculation
+
+```python
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+
+def calculate_previous_month():
+    """
+    Calculate the start and end dates of the previous month.
+    
+    Returns:
+        tuple: (start_date, end_date) in YYYY-MM-DD format
+        
+    Examples:
+        - If today is 2024-01-15, returns ('2023-12-01', '2023-12-31')
+        - If today is 2024-03-20, returns ('2024-02-01', '2024-02-29')
+    """
+    today = datetime.now()
+    
+    # First day of current month
+    first_of_month = today.replace(day=1)
+    
+    # Last day of previous month = day before first of current month
+    last_day_prev = first_of_month - timedelta(days=1)
+    
+    # First day of previous month
+    first_day_prev = last_day_prev.replace(day=1)
+    
+    return (
+        first_day_prev.strftime('%Y-%m-%d'),
+        last_day_prev.strftime('%Y-%m-%d')
+    )
+```
+
+### Data Collection Strategy
+
+#### 1. Players and Teams (Simple)
+- Direct call to existing `lib.player.all()` and `lib.team.all()`
+- No additional processing needed
+
+#### 2. Games Collection (Complex)
+**Challenge:** Need games from all teams without duplication
+
+**Solution:**
+```python
+def fetch_games_data(date_from, date_to, output_path):
+    """
+    Fetch all games in the date range.
+    
+    Strategy:
+    1. Get all teams using lib.team.all()
+    2. For each team, use find_games_by_team_and_date()
+    3. Collect all game records
+    4. Deduplicate by GAME_ID (games appear twice, once per team)
+    5. Write unique games to CSV
+    """
+    # Get all teams
+    teams_df = teams.get_teams()
+    
+    all_games = []
+    for team in teams_df.itertuples():
+        # Find games for this team
+        games_df = find_games_by_team_and_date(
+            team.id, 
+            date_from, 
+            date_to
+        )
+        all_games.append(games_df)
+        time.sleep(0.6)  # Rate limiting
+    
+    # Combine and deduplicate
+    combined = pd.concat(all_games, ignore_index=True)
+    unique_games = combined.drop_duplicates(subset=['GAME_ID'])
+    
+    write_csv(unique_games, output_path)
+    return unique_games
+```
+
+**Alternative (More Efficient):**
+Use `find_games_by_date()` for each day in the range, which is more efficient but requires iterating through dates.
```

Comment:
I think I would prefer to use this util for each day in the month to get games. From there, we can get box scores to see player and team stats